### PR TITLE
Show grants for user statement

### PR DESCRIPTION
**Pull request by n1tr0g**
As described https://github.com/influxdb/influxdb/issues/2212 we need to add show grants for user statement, so we are able to see user privileges.

### CODE REVIEW

File: server_test.go [hunk 1]
```diff
@@ -570,6 +570,35 @@ func TestServer_NonExistingUsers(t *testing.T) {
...
+	defer s.Close()
+
+	// Create some users.
+	s.CreateUser("susy", "pass", false)
+
+	// Ask for users data.
+	user := s.User("susy")
+
+	// Grant normal user write privileges on database "bar".
+	user.Privileges["bar"] = influxql.WritePrivilege
```
Comment (corylanou) [inline server_test.go]: Should you check later on that this privilege actually comes back?

Comment (n1tr0g) [inline server_test.go]: haha probably :+1: Will fix it now ...

**Comment by corylanou** [comment]
+1.  Can you also update CHANGELOG.md, and if you haven't already, please sign the [CLA](http://influxdb.com/community/cla.html)

File: influxql/parser.go [hunk 2]
```diff
@@ -1068,6 +1070,26 @@ func (p *Parser) parseShowServersStatement() (*ShowServersStatement, error) {
 }
 
 // parseShowDatabasesStatement parses a string and returns a ShowDatabasesStatement.
+// This function assumes the "SHOW GRANTS" tokens have already been consumed.
+func (p *Parser) parseGrantsForUserStatement() (*ShowGrantsForUserStatement, error) {
```
Comment (dgnorton) [inline influxql/parser.go]: Need to fix the comment for this method.

File: server_test.go [hunk 3]
```diff
@@ -570,6 +570,37 @@ func TestServer_NonExistingUsers(t *testing.T) {
 	}
 }
 
+// Ensure the server does return valid users grant data
+func TestServer_GetGrantPrivileges(t *testing.T) {
```
Comment (dgnorton) [inline server_test.go]: We're migrating tests to `cmd/influxd/server_integration_test.go`.  Can you remove this one and add tests to the table in that file?  The table is in the `runTestsData` func.

**Comment by dgnorton** [comment]
@n1tr0g  thanks!  Looks good overall.  Just a couple small changes that need to be made.

**Comment by n1tr0g** [comment]
No worries. I will apply the suggested changes tonight.

**Comment by n1tr0g** [comment]
Changed applied.

**Comment by toddboom** [comment]
@n1tr0g we just merged some major changes for rc32 - could you rebase this? thanks!

**Comment by n1tr0g** [comment]
hehe that's really a MAJOR change. I will have a look.

**Comment by n1tr0g** [comment]
merged and code modified. Please review.

**Comment by n1tr0g** [comment]
Good work with the release of 0.9.0, but I am surprised that this change didn't get merged before that. 

Is there an alternative way to see users privileges on a database ?

**Comment by corylanou** [comment]
@n1tr0g I take the blame for not merging this on time.  Sorry about that, I know we created unnecessary work for you, and that's not cool.  Looks like it might need a rebase one more time likely due to a changelog conflict if I had to guess, but I also know we've already asked you to do more work than you should of.  If you want, I can rebase it for you and merge it in.

Thanks for all your work, and so sorry this didn't get merged.

**Comment by n1tr0g** [comment]
HeHe yeah this PR is taking ages, but no problem. Rebase done.

**Comment by dgnorton** [comment]
+1 LGTM
@n1tr0g, thanks for sticking with it!

**Comment by otoolep** [comment]
+1 here too -- I'll merge now since @dgnorton also approved.

**Comment by corylanou** [comment]
Wohoo its merged!  Thanks again @n1tr0g!!!

**Comment by n1tr0g** [comment]
haha now I can have a drink to celebrate that :)

**Pull request merged**